pkg/steps/image_stream_util.go

```diff
@@ -0,0 +1,101 @@
+package steps
+
+import (
+	"encoding/json"
+	"fmt"
+
+	imageapi "github.com/openshift/api/image/v1"
+	imageclientset "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1"
+	coreapi "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/retry"
+)
+
+func createImageStreamWithTag(
+	isClient imageclientset.ImageStreamsGetter,
+	istClient imageclientset.ImageStreamTagsGetter,
+	is *imageapi.ImageStream,
+	ist *imageapi.ImageStreamTag,
+	dry bool,
+) error {
+	if dry {
+		isJSON, err := json.MarshalIndent(is, "", "  ")
+		if err != nil {
+			return fmt.Errorf("failed to marshal imagestream: %v", err)
+		}
+		fmt.Printf("%s\n", isJSON)
+
+		istJSON, err := json.MarshalIndent(ist, "", "  ")
+		if err != nil {
+			return fmt.Errorf("failed to marshal imagestreamtag: %v", err)
+		}
+		fmt.Printf("%s\n", istJSON)
+		return nil
+	}
+
+	_, err := isClient.ImageStreams(is.Namespace).Get(is.Name, meta.GetOptions{})
+	if errors.IsNotFound(err) {
+		_, err = isClient.ImageStreams(is.Namespace).Create(is)
+	}
+	if err != nil {
+		return fmt.Errorf("could not retrieve target imagestream: %v", err)
+	}
```

Comment:
I wondered this myself, but this logic is what's always existed. The thing to note is, if there's a conflict for the imagestream, it's fine, since the imagestream is just a container of image stream tags, so in the end, as long as it exists; we're happy (that's not true for tags however, for tags the value matters).

Reply:
Yeah, that's exactly why I would expect `Create` and ignore `AlreadyExists`. The `Get` followed by `Create` looks risky...

@stevekuznetsov WDYT?

Reply:
Yeah why not just Create and ignore the AlreadyExists? I know you copy pasta this but we might as well fix the code. I probably wrote it poorly a year ago :)

Reply:
I can definitely change it.